src/utils/instrumentation.ts

```diff
@@ -64,7 +69,7 @@ export function populateInstrumentationInfo(
             validateAndUpdateInstrumentation(info);
           // Indicate that instrumentation info log entry already exists
           // and that current library info was added to existing log entry
-          isInfoAdded = isWritten = true;
+          global.instrumentationWritten = isInfoAdded = isWritten = true;
```

Comment:
There are 2 cases when instrumentation can be added:

1. By higher level library (e.g. `nodejs-logging-bunyan` or `nodejs-logging-winston`) - in this case we need to inspect the payload and see if we have a diagnostic record available so we can add to existing record the info about `nodejs-logging`.
2. By `nodejs-logging` library itself if it is used directly - in this case the new record will be appended.

So below explanation per flag - please let me know what can be done to make it more clear/efficient:

1. `global.instrumentationAdded` tracks if instrumentation should be added or not - it is a global flag tracking instrumentation status across libraries. Please note that nodejs-logging-bunyan` or `nodejs-logging-winston` set it to `true` before writing instrumentation record, so by the time it gets to `nodejs-logging` this flag will be already `false` - thats a reason we inspect a payload for instrumentation in `nodejs-logging`.
2. `global.instrumentationWritten` is just another indication if we should do any payload inspection for instrumentation or not. 
3. `isWritten` basically a copy of `global.instrumentationAdded` before it was set to `true` in `setInstrumentationStatus()`
4. `isInfoAdded` is tracking if any instrumentation data was added - if yes, it is returned to a caller so the partialSuccess will be added to write request

I will rename the `global.instrumentationWritten` - please let me know if that makes sense

Reply:
Ok, I approved it as-is. It still seems a bit clunky though, so it might be worth re-thinking it. Maybe see if we can get away with one variable tracking the instrumentation state instead of multiple flags?

Reply:
Yeah, I must admit this is indeed seems clunky. Thats said, given a fact for so many dependencies I preferred to keep multiple variables so it would be easier to track different states of a logic. Lets see if we will have any issues with that approach in the future - then we could rework it